monai/data/box_utils.py

```diff
@@ -0,0 +1,291 @@
+        (xmin, ymin, xmax, ymax) or (xmin, ymin, zmin, xmax, ymax, zmax)
+    Returns:
+        bool, whether the box is valid
+    Example:
+        corners = (torch.ones(10,1), torch.ones(10,1), torch.ones(10,1), torch.ones(10,1))
+        check_corner(corners) will return True
+    """
+    spatial_dims = get_dimension(corners=corners)
+    box_error = corners[spatial_dims] < corners[0]
+    for axis in range(1, spatial_dims):
+        box_error = box_error | (corners[spatial_dims + axis] < corners[axis])
+    if box_error.sum() > 0:
+        return False
+    else:
+        return True
+
+
+def convert_box_mode(
+    boxes: NdarrayOrTensor,
+    src_mode: Union[str, BoxMode, Type[BoxMode], None] = None,
+    dst_mode: Union[str, BoxMode, Type[BoxMode], None] = None,
+) -> NdarrayOrTensor:
+    """
+    This function converts the boxes in src_mode to the dst_mode
+    Args:
+        boxes: source bounding box, Nx4 or Nx6 torch tensor or ndarray
+        src_mode: source box mode. If mode is not given, this func will assume mode is StandardMode.
+        dst_mode: target box mode. If mode is not given, this func will assume mode is StandardMode.
+        src_mode and dst_mode can be:
+            #. str: choose from monai.utils.enums.BoundingBoxMode, including
+                "xyxy": boxes has format [xmin, ymin, xmax, ymax]
+                "xyzxyz": boxes has format [xmin, ymin, zmin, xmax, ymax, zmax]
+                "xxyy": boxes has format [xmin, xmax, ymin, ymax]
+                "xxyyzz": boxes has format [xmin, xmax, ymin, ymax, zmin, zmax]
+                "xyxyzz": boxes has format [xmin, ymin, xmax, ymax, zmin, zmax]
+                "xywh": boxes has format [xmin, ymin, xsize, ysize]
+                "xyzwhd": boxes has format [xmin, ymin, zmin, xsize, ysize, zsize]
+                "ccwh": boxes has format [xcenter, ycenter, xsize, ysize]
+                "cccwhd": boxes has format [xcenter, ycenter, zcenter, xsize, ysize, zsize]
+            #. BoxMode class: choose from
+                CornerCornerModeTypeA: equivalent to "xyxy" or "xyzxyz"
+                CornerCornerModeTypeB: equivalent to "xxyy" or "xxyyzz"
+                CornerCornerModeTypeC: equivalent to "xyxy" or "xyxyzz"
+                CornerSizeMode: equivalent to "xywh" or "xyzwhd"
+                CenterSizeMode: equivalent to "ccwh" or "cccwhd"
+            #. BoxMode instance
+            #. None: will assume mode is StandardMode
+    Returns:
+        boxes_dst: bounding box with target mode, does not share memory with original boxes
+
+    Example:
+        boxes = torch.ones(10,6)
+        box_convert_mode(boxes=boxes, src_mode="xyzxyz", dst_mode=CenterSizeMode)
+    """
+
+    # if mode not changed, return original box
+    src_boxmode = get_boxmode(src_mode)
+    dst_boxmode = get_boxmode(dst_mode)
+    if type(src_boxmode) is type(dst_boxmode):
+        return deepcopy(boxes)
```

Comment:
May I know why `deepcopy` here?

Thanks.

Reply:
I was trying to make sure the returned boxes does not share memory with input boxes

Reply:
OK, I just feel it conflicts with your above comment: "if mode not changed, return original boxes", maybe you can change the comment to "if mode not changed, deepcopy the original boxes".

Thanks.